Unstable test `dump_test.TestDumpBlock`

```
=== RUN   TestDumpBlock
    dump_test.go:62:
        	Error Trace:	dump_test.go:62
        	Error:      	Target error should be in err chain:
        	            	expected: "canceling query due to user request"
        	            	in chain: "context canceled"
        	Test:       	TestDumpBlock
--- FAIL: TestDumpBlock (0.00s)
=== RUN   TestDumpBlock
    dump_test.go:62:
        	Error Trace:	dump_test.go:62
        	Error:      	Target error should be in err chain:
        	            	expected: "canceling query due to user request"
        	            	in chain: "context canceled"
        	Test:       	TestDumpBlock
--- FAIL: TestDumpBlock (0.00s)
=== RUN   TestDumpBlock
```

--- a/dumpling/export/dump_test.go
+++ b/dumpling/export/dump_test.go
@@ -53,14 +53,16 @@ func TestDumpBlock(t *testing.T) {
 		time.Sleep(time.Second)
 		return context.Canceled
 	})
+
 	writerCtx := tctx.WithContext(writingCtx)
 	// simulate taskChan is full
 	taskChan := make(chan Task, 1)
 	taskChan <- &TaskDatabaseMeta{}
 	d.conf.Tables = DatabaseTables{}.AppendTable(database, nil)
 	d.conf.ServerInfo.ServerType = version.ServerTypeMySQL
-	require.ErrorIs(t, d.dumpDatabases(writerCtx, baseConn, taskChan), sqlmock.ErrCancelled)
 	require.ErrorIs(t, wg.Wait(), writerErr)
+	// if writerCtx is canceled , QuerySQL in `dumpDatabases` will return sqlmock.ErrCancelled
+	require.ErrorIs(t, d.dumpDatabases(writerCtx, baseConn, taskChan), sqlmock.ErrCancelled)
 }
 
 func TestDumpTableMeta(t *testing.T) {

